src/bare_item.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{Num, Parser};
+use std::{convert::TryFrom, fmt, ops::Deref};
+
+// TODO: Necessary? rust_decimal seems to validate already
+// pub struct Decimal(rust_decimal::Decimal);
+
+/// Integers have a range of -999,999,999,999,999 to 999,999,999,999,999 inclusive (i.e., up to fifteen digits, signed), for IEEE 754 compatibility.
+///
+/// The ABNF for Integers is:
+/// ```abnf,ignore,no_run
+/// sf-integer = ["-"] 1*15DIGIT
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub struct Integer(i64);
+
+impl Deref for Integer {
+    type Target = i64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl TryFrom<i64> for Integer {
+    type Error = &'static str;
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        let input_string = value.to_string();
+        let mut input_chars = input_string.chars().peekable();
+        let validated = Parser::parse_number(&mut input_chars)?;
+        match validated {
+            Num::Integer(val) => Ok(Integer(val)),
+            Num::Decimal(_) => Err("Input is Decimal, expected Integer"),
+        }
+    }
+}
+
+impl fmt::Display for Integer {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+// TODO: how to get around naming collision without using std::string::String everywhere?
```

Comment:
If you have an idea here, that would be appreciated. The name `BareItemString` is not really ergonomic 😆

Reply:
@undef1nd Was wondering if we should call them all BareItemInteger,.. etc for consistency 🤷

Reply:
With your proposal of constructing these values only internally and exposing them only via constructor methods, I think that would be the right call.

Reply:
Renamed them all for now. If we decide to not expose them at all outside the `bare_item` module, we could technically rename back as it would then just be redundant. Either way no harm caused 🤷‍♂️